extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/devconsole/DevConsoleProcessor.java

```diff
@@ -36,57 +36,79 @@ public DevConsoleTemplateInfoBuildItem collectAdditionalEndpoints(
         return new DevConsoleTemplateInfoBuildItem("additionalEndpointInfo", additionalEndpoint);
     }
 
-    // knownPaths variable contain set of static file resources that are available from classpath
     @BuildStep(onlyIf = IsDevelopment.class)
     public DevConsoleTemplateInfoBuildItem collectStaticResourcesInfo(
             ApplicationArchivesBuildItem applicationArchivesBuildItem) throws Exception {
 
-        Set<String> knownPaths = new TreeSet<>();
+        StaticResourceInfo staticResourceInfo = new StaticResourceInfo();
+
         for (ApplicationArchive i : applicationArchivesBuildItem.getAllApplicationArchives()) {
             Path resource = i.getChildPath(StaticResourcesRecorder.META_INF_RESOURCES);
             if (resource != null && Files.exists(resource)) {
-                collectKnownPaths(resource, knownPaths);
+                collectKnownPaths(resource, staticResourceInfo);
             }
         }
 
         ClassPathUtils.consumeAsPaths(StaticResourcesRecorder.META_INF_RESOURCES, resource -> {
-            collectKnownPaths(resource, knownPaths);
+            collectKnownPaths(resource, staticResourceInfo);
         });
-
-        return new DevConsoleTemplateInfoBuildItem("staticResourcesInfo", knownPaths);
+        return new DevConsoleTemplateInfoBuildItem("staticResourcesInfo", staticResourceInfo);
     }
 
-    private void collectKnownPaths(Path resource, Set<String> staticResources) {
+    private void collectKnownPaths(Path resource, StaticResourceInfo staticResourceInfo) {
+
         try {
             Files.walkFileTree(resource, new SimpleFileVisitor<Path>() {
 
+                final Stack<StaticResourceInfo.StaticFile> stack = new Stack<>();
+
                 @Override
-                public FileVisitResult visitFile(Path p, BasicFileAttributes attributes) throws IOException {
-                    String simpleName = p.getFileName().toString();
-                    String file = resource.relativize(p).toString();
-                    if (simpleName.equals("index.html") || simpleName.equals("index.htm")) {
-                        Path parent = resource.relativize(p).getParent();
-                        if (parent == null) {
-                            staticResources.add("/");
-                        } else {
-                            String parentString = parent.toString();
-                            if (!parentString.startsWith("/")) {
-                                parentString = "/" + parentString;
-                            }
-                            staticResources.add(parentString + "/");
-                        }
+                public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+
+                    Path relativeFile = resource.relativize(dir);
+                    String relativeFileString = relativeFile.toString();
+                    relativeFileString = "/" + relativeFileString;
```

Comment:
You can merge the two previous lines for better clarity.

Reply:
Sure